### PR TITLE
Inventory search

### PR DESCRIPTION
**Pull request by noahboegli**
Implements
- https://github.com/PDG-2023/Project/issues/56

Search endpoint is `/inventory/{id}/search&searchTerm=STRING`

Results are returned as in an array of the DTO below
```yaml
SearchResultDTO:
  type: object
  properties:
    id:
      type: integer
    name:
      type: string
    description:
      type: string
    entityType:
      type: string
```

Where `entityType` can be any of `location`, `user`, `itemModel`

Fields searched are description and name for locations and item models, firstName and lastName for users.

Added tests to back the implementation.

> __Warning__
> This branch is based on `dev/total` and not master, because reasons. Be cautious when merging.

### CODE REVIEW

**Comment by HugoMendes98** [comment]
@noahboegli I change the branch target, Github will automatically swap, once `dev/total` is merged and deleted

File: apps/backend/src/main/java/ch/heig/pdg/backend/services/InventoryService.java [hunk 1]
```diff
@@ -27,6 +35,41 @@ public List<InventoryDTO> getInventories(HugoSearchFilter<Inventory> filter) {
...
+                    SearchResultDTO dto = new SearchResultDTO();
+                    dto.id(i.getId());
+                    dto.entityType("itemModel");
+                    dto.description(i.getDescription());
+                    dto.name(i.getName());
+                    return dto;
+                }
+        );
+
+        return Stream.of(locations, users, itemModels).flatMap(o -> o).collect(Collectors.toList());
```
Comment (HugoMendes98) [inline apps/backend/src/main/java/ch/heig/pdg/backend/services/InventoryService.java:70]: Is there a reason to `flatMap` here?
Would'nt it be easier to return a dict of the results (`{"locations": [], "items-models": [], "users": []}`) and avoid the transformations above? 

> No special need to change

**Review by HugoMendes98 — APPROVED** [review]
Looks good for the current use.

> A "offset-n-limit" parameters could be needed on a next iteration, when there will be a lot of data

File: apps/backend/src/main/java/ch/heig/pdg/backend/services/InventoryService.java (hunk 1 above)
Comment (noahboegli) [reply apps/backend/src/main/java/ch/heig/pdg/backend/services/InventoryService.java:70]: It's to simplify the declaration in the api spec and to streamline the data, but I'm happy to change it if required.

**Review by nigelmann — APPROVED** [review]
Seems good to me !

**Pull request merged**